apps/hash-graph/lib/graph/src/store/validation.rs

```diff
@@ -24,26 +28,97 @@ use crate::{
     },
 };
 
-#[derive(Debug, Copy, Clone)]
+#[derive(Debug, Clone)]
+enum Access<T> {
+    Granted(T),
+    Denied,
+    Malformed,
```

Comment:
For what do we need these variants? Just as a small optimization in a multithreading context?

Reply:
> I mean if you have a `Denied` or `Malformed` you'll fail anyway, right?

Exactly, which is why we just "replay" that failure, should it happen.